otlp/otlp_request_builder.h

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Splunk Inc.
```

Comment:
Can use this to simplify OtlpGrpcFormatter::format:

` ExportMetricsServiceRequest request = 
    otlp_client::OtlpRequestBuilder()
      .metric(metric_name)
      .sum()
      .number_data_point(metric_value, labels, timestamp);
writer->write(request);
`

Reply:
Is the idea of the sum() streaming operator in this case, so that the builder can be extended for the other metric types later as desired?

Reply:
Yeah I'm trying to leave room for additional types (gauge, histogram, etc.) if we need to add later.  Right now sum is the only implemented one though.